File: CuentaBancaria/muestras_de_posibles_soluciones/java/CuentaBancaria.java
package CuentaBancaria.muestras_de_posibles_soluciones.java;

public class CuentaBancaria {
    
    private java.util.UUID codigoDeCuenta;
    private String titular;
    Double saldo;

    public CuentaBancaria(String titular) {
        codigoDeCuenta = java.util.UUID.randomUUID();
        this.titular = titular;
        saldo = 0.0;
    }

    public String getCodigoDeCuenta() {
        return codigoDeCuenta.toString();
    }

    public String getTitular() {
        return titular;
    }

    public Double getSaldo() {
        return saldo;
    }

    public void hacerIngreso(Double cantidad) {
        if (cantidad > 0) {
            saldo = saldo + cantidad;
        } else {
            throw new IllegalArgumentException("Se han de indicar siempre cantidades positivas.");
        }
    }

    public void hacerReintegro(Double cantidad) {
        if (cantidad > 0) {
            if (saldo > cantidad) {
                saldo = saldo - cantidad;
            } else {
                throw new IllegalArgumentException("El saldo es insuficiente para hacer este reintegro.");
            }
        } else {
            throw new IllegalArgumentException("Se han de indicar siempre cantidades positivas.");
        }
    }
    
}

File: CuentaBancaria/muestras_de_posibles_soluciones/python/cuenta_bancaria.py
from uuid import uuid4


class Cuenta:
    def __init__(self, titular):
        self._codigoDeCuenta = uuid4()
        self._titular = titular
        self._saldo = 0.0

    def getCodigoDeCuenta(self):
        return self._codigoDeCuenta

    def getTitular(self):
        return self._titular

    def getSaldo(self):
        return self._saldo

    def hacerIngreso(self, cantidad):
        if cantidad > 0:
            self._saldo = self._saldo + cantidad
        else:
            raise ValueError("Se ha de indicar siempre una cantidad positiva.")

    def hacerReintegro(self, cantidad):
        if cantidad > 0:
            if self._saldo >= cantidad:
                self._saldo = self._saldo - cantidad
            else:
                raise ValueError(
                    f"El saldo es insufiente para hacer este reintegro de {cantidad}."
                )
        else:
            raise ValueError("Se ha de indicar siempre una cantidad positiva.")
